md/api.py

```diff
@@ -44,7 +43,10 @@ def query(self, results, group_by, filter_=None):
             data = {}
             if 'year' in group_by:
                 data['year'] = stop['year'].year
-            # XXX check for 'purpose' here after ODPM-31 is delivered.
+            if 'purpose' in group_by:
+                purpose = dict(PURPOSE_CHOICES).get(stop['purpose'],
+                                              stop['purpose'])
```

Comment:
this code should line up with something, such as the stop['purpose'] on the line above  (similarly for the existing code a few lines below)